osu.Game.Tests/Visual/Online/TestSceneChatOverlayV2.cs

```diff
@@ -468,21 +467,26 @@ public void TestKeyboardNextChannel()
                 chatOverlay.Show();
             });
 
-            AddAssert("Channel 1 displayed", () => channelIsVisible && currentDrawableChannel.Channel == testChannel1);
-
+            waitForChannel1Visible();
             AddStep("Press document next keys", () => InputManager.Keys(PlatformAction.DocumentNext));
-            AddAssert("Channel 2 displayed", () => channelIsVisible && currentDrawableChannel.Channel == testChannel2);
+            waitForChannel2Visible();
 
             AddStep("Press document next keys", () => InputManager.Keys(PlatformAction.DocumentNext));
-            AddAssert("PM Channel 1 displayed", () => channelIsVisible && currentDrawableChannel.Channel == pmChannel1);
+            AddUntilStep("PM Channel 1 displayed", () => channelIsVisible && currentDrawableChannel?.Channel == pmChannel1);
 
             AddStep("Press document next keys", () => InputManager.Keys(PlatformAction.DocumentNext));
-            AddAssert("PM Channel 2 displayed", () => channelIsVisible && currentDrawableChannel.Channel == pmChannel2);
+            AddUntilStep("PM Channel 2 displayed", () => channelIsVisible && currentDrawableChannel?.Channel == pmChannel2);
 
             AddStep("Press document next keys", () => InputManager.Keys(PlatformAction.DocumentNext));
-            AddAssert("Channel 1 displayed", () => channelIsVisible && currentDrawableChannel.Channel == testChannel1);
+            waitForChannel1Visible();
         }
 
+        private void waitForChannel1Visible() =>
```

Comment:
Could this be generalised to `void waitForChannelVisible(Channel channel)` since there are other test cases checking against PM and announce visibility which may also fail?

Reply:
They already use `AddUntilStep`. Generalising isn't so easy, due to the message and requiring a delegate function input to get the channel correctly.

Reply:
I see. I had naively assumed you could generalise the channel test function like this but this seems to not work at all :thinking: 
```csharp
void waitForChannelVisible(Channel channel) =>
    AddUntilStep($"Channel {channel.Name} visible", () => channelIsVisible && currentDrawableChannel?.Channel == channel);
```

Reply:
It doesn't work because in some cases the channel is created inside the schedule in `[SetUp]`, so sometimes the proposed method is called before the channels are initialised. You'd have to pass a `Func<Channel>` as mentioned above, and at that point I'm not sure it's worth the trouble.